auditlog/models.py

```diff
@@ -208,6 +215,65 @@ def _get_pk_value(self, instance):
             pk = self._get_pk_value(pk)
         return pk
 
+    def _get_serialized_data_or_none(self, instance):
+        from .registry import auditlog
+
+        opts = auditlog.get_serialize_options(instance.__class__)
+        if not opts["serialize_data"]:
+            return None
+
+        kwargs = opts.get("serialize_kwargs", {})
+        kwargs.setdefault("fields", None)
+        data = dict(json.loads(serializers.serialize("json", (instance,), **kwargs))[0])
+        model_fields = auditlog.get_model_fields(instance.__class__)
+
+        if opts["serialize_auditlog_fields_only"]:
+            data = self._filter_serialized_fields_to_eligible(data, model_fields)
```

Comment:
Isn't it possible to check this option before serializing the data? I mean by adding the fields to `kwarg`? Then we would have one less loop over all data.

Reply:
@hramezani it sure is! Good thinking. Sometimes I can't see the forest from the trees when I'm moving code around. Check it out now.